wasmi_v1/src/lib.rs

```diff
@@ -3,6 +3,7 @@
 //! These closely mirror the WebAssembly specification definitions.
 //! The overall structure is heavily inspired by the `wasmtime` virtual
 //! machine architecture.
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
I cannot believe we forgot about this all the time in `wasmi_v1` ...

Reply:
For something that was not tested on no_std it held up remarkably well :-)

(Then again, now that there is alloc, it has become easier ... an alloc-less version would be really great to have, but seems not to be trivial from what I've seen when inserting the `alloc::`s).

Reply:
What use cases do you have for an alloc less version? The difference between alloc and no alloc is a global allocator which could be as simple as a trivial bump allocator such as bumpalo.

Reply:
Better predictability on an embedded system -- plus it's hard to determine whether a plain bump allocator is really sufficient for a project like this or not, and a good allocator is costly on these systems.

Predictability is especially a concern when there are multiple components "like this" on a device. While it's only one (and if it's really "bump allocator friendly"), I can pick a simple allocator and worst that happens is that my WASM gets stuck out of memory. But once I have, say, a WASM and a Bluetooth stack that "only" need a simple allocator, suddenly a rogue WASM program can lock me out of Bluetooth. (Same can go for two WASMI instances, when one is performing administrative and another user provided tasks).

I think that for some uses of alloc in here it might even be simple to solve (eg. by compatibly turning `CallStack` into `CallStack<Frames: VecLikesh<FunctionFrame> = Vec<FunctionFrame>>`, which I'd then rather instanciate as `CallStack<heapless::Vec<FunctionFrame, 16>>` or so), but around the function maps or the WASM parser I have no clue whether the same would be practical, or (given there is a parser present) whether WASMI even makes sense on these platforms (on which I'd prefer to interpret WASM in-place or at least from something that gets built once from WASM input).

Reply:
Thanks for the clarification.
However, in all honesty I do not plan on supporting `wasmi` without `alloc` since that would be an absolut epic amount of additional work. Instead I'd really try to use a suiting heap allocator and maybe make use of Wasm `multiple-memories` once it has been standardized so that different Wasm instances do not compete for linear memory.